ngraph/core/src/function.cpp

```diff
@@ -102,6 +161,58 @@ void Function::check_all_parameters_registered() const
                            unregistered_parameters.str());
 }
 
+void Function::check_all_variables_registered(
+    const std::vector<shared_ptr<Node>>& ordered_ops) const
+{
+    OV_ITT_SCOPED_TASK(ngraph::itt::domains::nGraphPass_LT,
+                       "Function::check_all_variables_registered");
+    std::stringstream unregistered_variables;
+    for (auto& node : ordered_ops)
+    {
+        const auto& variable_op = dynamic_pointer_cast<VariableExtension>(node);
+        if (variable_op &&
+            std::find(m_variables.begin(), m_variables.end(), variable_op->get_variable()) ==
+                m_variables.end())
+            unregistered_variables << variable_op->get_variable_id() << std::endl;
+    }
+    if (!unregistered_variables.str().empty())
+        throw ngraph_error("Function references undeclared variables: " +
+                           unregistered_variables.str());
+}
+
+void Function::auto_detect_variables(const std::vector<std::shared_ptr<Node>>& ordered_ops)
+{
+    unordered_set<VariablePtr> variables;
+    for (const auto& op : ordered_ops)
+    {
+        if (const auto& variable_op = dynamic_pointer_cast<VariableExtension>(op))
+        {
+            variables.insert(variable_op->get_variable());
+        }
+    }
+    m_variables.assign(variables.begin(), variables.end());
+}
+
+void Function::auto_detect_parameters(const std::vector<std::shared_ptr<Node>>& ordered_ops)
+{
+    for (const auto& op : ordered_ops)
+    {
+        if (const auto& param = dynamic_pointer_cast<opset7::Parameter>(op))
+        {
+            m_parameters.push_back(param);
+        }
+    }
+}
+
+void Function::prerequirements(bool detect_variables, bool detect_parameters)
```

Comment:
Can we add documentation for this method?

Reply:
done